packages/web3-eth/src/utils/transaction_builder.ts

```diff
@@ -166,15 +166,15 @@ export async function defaultTransactionBuilder<ReturnType = Record<string, unkn
 			data: populatedTransaction.data,
 			input: populatedTransaction.input,
 		});
-	} else if (!isNullish(populatedTransaction.input)) {
-		populatedTransaction.data = populatedTransaction.input;
-		delete populatedTransaction.input;
+	} else if (!isNullish(populatedTransaction.data)) {
+		populatedTransaction.input = populatedTransaction.data;
+		delete populatedTransaction.data;
 	}
 
-	if (isNullish(populatedTransaction.data) || populatedTransaction.data === '') {
-		populatedTransaction.data = '0x';
-	} else if (!populatedTransaction.data.startsWith('0x')) {
-		populatedTransaction.data = `0x${populatedTransaction.data}`;
+	if (isNullish(populatedTransaction.input) || populatedTransaction.input === '') {
+		populatedTransaction.input = '0x';
+	} else if (!populatedTransaction.input.startsWith('0x')) {
+		populatedTransaction.input = `0x${populatedTransaction.input}`;
```

Comment:
Can you add unit test for newly added code. Thanks.

Reply:
Yes, this wasn't meant to be marked as ready for review - need to update changelog and migration guide too

Reply:
Added in [this](https://github.com/web3/web3.js/pull/5915/commits/90e708adaaaa359bcf3a2055fa39bae5dbe80b99) commit